Implement_A_Stack/myDataStructs.py

```diff
@@ -1,102 +1,94 @@
 # -*- coding: utf-8 -*-
-"""
-Spyder Editor
-
-This is a temporary script file.
-"""
 import copy
 
-class myStack(object):
-    'My stack implementation'
+class MyStack(object):
+    """My stack implementation"""
+
     def __init__(self):
-        self.contents = ()
+        self.__contents = ()
 
 
     def empty(self):
-        if len(self.contents):
-            return False
-        else:
-            return True
+        return len(self.__contents) == 0
 
 
     def push(self, val):
-        self.contents = (val, self.contents)
+        self.__contents = (val, self.__contents)
 
 
     def pop(self):
         try:
-            val = self.contents[0]
-            self.contents = self.contents[1]
+            val = self.__contents[0]
+            self.__contents = self.__contents[1]
             return val
         except:
             return None
 
 
     def peek(self):
-        val = self.pop()
-        self.push(val)
-        return val
+        try:
+            return self.__contents[0]
+        except:
+            return None
 
 
     def search(self, match_val):
         current_loc = 0
         match_loc = -1
         val = None
-        starting_state = copy.deepcopy(self.contents)
-
-        while True:
-            if self.empty():
-                break
+        starting_state = copy.deepcopy(self.__contents)
 
+        while not self.empty():
             current_loc += 1
             val = self.pop()
 
             if val == match_val:
                 match_loc = current_loc
                 break
 
-        self.contents = starting_state
-
+        self.__contents = starting_state
         return match_loc
 
+
     def size(self):
```

Comment:
Copying is space inefficient, and iterating twice (once for the copy) is time inefficient. First thought is to keep a `self.size` counter and increment on every `push`/decrement on every `pop`.